File: TaskOOP/department.py
from person.human import Human
from person.manager import Manager


class Department(Human):
    def __init__(self):
        self.managers = list(Manager.managers)
        pass

    def manList(self):
        print("Manager list:")
        for i in range(len(self.managers)):
            print(self.managers[i].sName)
        print('\n')

    def salary(self):
        print("Salary Humans:")
        for i in range(len(Human.humans)):
            print("{0} {1}, got salary: {2}".format(Human.humans[i].firstName, Human.humans[i].sName, Human.humans[i].sal()))
        print('\n')

File: OOPJava/src/main/java/exercise1/Department.java
package exercise1;

import exercise1.person.Employee;
import exercise1.person.Manager;

import java.util.ArrayList;
import java.util.List;

public class Department {
    List<Manager> managerList = new ArrayList<Manager>();

    public void setManagerList(Manager...managers) {
        if (managers.length == 0) {
            System.out.println("Departament need at last 1 employee");
        }
        for (Manager i : managers) {
            managerList.add(i);
        }
    }

    public void giveSalary(){
        for (Manager i : managerList) {
            System.out.println(i.firstName + " " + i.secondName +" " +": got a salary: " + i.getSalary());
            i.giveSalaryTeam();
        }

    }
}
